### DISCUSSION TITLE
有没有可能性通过扩展源码导出replace后的PAG FILE

### DISCUSSION THREAD

**Discussion by mochengdx** [Tencent/libpag#2192] [General] [answered]
@齐国鹏 你好，咨询一个问题，背景是我们有一套基于PAG的云端渲染方案，大概思路是  在replace 后 并基于 canvas播放并录制视频，因为 这套方案是基于 chrome 内核，录制视频导致CPU过高，目前在进一步寻找优化方案，其中一个重要方案是 需要导出替换素材后的PAG视频，再由服务端转码（不用canvas录），当前想确认下这个方案是否可行，在目前提供的接口中 ，比如 PAGFile中并没有类似实现，是否有有可能扩展这个能力，比如我们自己基于源码扩展并导出PAG FILE ，或是其它的一些思路，请指教

**Comment by kevingpqi123** [comment, marked as answer]
1、如果需要 web 侧渲染导出视频，由于性能的限制，确实不适合基于 web 平台的接口导出视频，常用的做法都是在服务端渲染导出视频然后 web 页面分发，比如腾讯 AMS 的视频广告以及一些运营活动都是这样的处理思路。
2、既然 web 平台可以替换素材，服务端同样也可以替换素材，因此后面一部分就是一个伪需求，你们把所有需要替换修改的内容一起发送给服务端，由服务端替换渲染导出视频即可，web 页面仅仅是一个交互入口。

**Reply by mochengdx** [reply]
所以怎么通过服务端来进行 素材替换，目前来说 只提供 IOS ANDRIOD 和 WEB

**Reply by mochengdx** [reply]
@kevingpqi123 局限与对于 Lib 的了解，是否可能通过 当前开源的场景 做一套 SDK，类似与直接在服务端加载 pag 模板，replace 素材、导出替换PAG文件、编解码 ，请帮忙评估下可行性 及 实际的方案；目前我能想到的策略是 在 centos 安装 android 环境，调用 android 替换，再导出这样；其实核心的问题有没有类似JAR包或是C++库直接load  、 replace 、 export ，然后由我们文件系统编解码

**Reply by kevingpqi123** [reply]
建议系统看下 pag 的官方文档：https://pag.art/docs/sdk.html#%E6%A1%8C%E9%9D%A2%E7%AB%AF%E6%8E%A5%E5%85%A5
如果你对于服务端开发不太清楚，建议咨询相关有开发经验的同学
另外，我们只解答 pag 使用相关的内容，涉及到视频后编辑、方案评估等其它内容不在我们技术支持的范围之内
